### PR TITLE
Fix view refreshing when using tabs

### PR DESCRIPTION
**Pull request by KillariDev**
fix https://github.com/DarkFlorist/TheInterceptor/issues/492
when tab is refreshed it loses it's state. This PR makes it so that on refresh the tab will request the data from storage again. This system is added to all popup views that we have (addressbook works with refresh)

### CODE REVIEW

File: app/ts/components/App.tsx
Comment (MicahZoltu) [inline app/ts/components/App.tsx:41]: Still resisting `Signal`s?

File: app/ts/components/pages/InterceptorAccess.tsx
Comment (MicahZoltu) [inline app/ts/components/pages/InterceptorAccess.tsx:201]: `[]` as second parameter to `useEffect` means it will run once when the component is first instantiated.  What does missing parameter mean (like the previous `useEffect`)?  The docs don't seem to specify.  Is it the same as `[]`?  If so, we should probably be consistent on which one we use.

**Review by MicahZoltu — APPROVED** [review]
This is a fairly large change and I only gave it a cursory review.  Recommend getting another set of eyes on it before merging.

File: app/ts/components/App.tsx
Comment (KillariDev) [reply app/ts/components/App.tsx:41]: no, but I feel it requires bigger change to port over :/

File: app/ts/components/pages/InterceptorAccess.tsx
Comment (KillariDev) [reply app/ts/components/pages/InterceptorAccess.tsx:201]: they are not the same, [] means it runs once, and missing parameter means its run everytime the component refreshes

Comment (KillariDev) [reply app/ts/components/pages/InterceptorAccess.tsx:201]: you can read about that eg in here: <https://stackoverflow.com/questions/55240526/useeffect-second-argument-variations-in-react-hook>

File: app/ts/background/windows/confirmTransaction.ts
Comment (jubalm) [inline app/ts/background/windows/confirmTransaction.ts:111]: can use a shorthand property

File: app/ts/background/windows/interceptorAccess.ts
Comment (jubalm) [inline app/ts/background/windows/interceptorAccess.ts:186]: why the change from snake case to camel case on this method?

File: app/ts/background/windows/personalSign.ts
Comment (jubalm) [inline app/ts/background/windows/personalSign.ts:39]: looks like not all conditions are returned from this function?

File: app/ts/components/subcomponents/ChainSelector.tsx
Comment (jubalm) [inline app/ts/components/subcomponents/ChainSelector.tsx:35]: this feels like a good place for a button element instead of anchor

File: app/ts/background/windows/interceptorAccess.ts
Comment (KillariDev) [reply app/ts/background/windows/interceptorAccess.ts:186]: the idea is to keep `popup_` prefix and then move to camelcase for all messages. As they are currently quite mixed. Not sue what is the best strategy with these 🤔

File: app/ts/background/windows/personalSign.ts
Comment (KillariDev) [reply app/ts/background/windows/personalSign.ts:39]: it returns void so it doesn't need to return in all conditions

File: app/ts/components/subcomponents/ChainSelector.tsx
Comment (KillariDev) [reply app/ts/components/subcomponents/ChainSelector.tsx:35]: hmm, you feel the styling of button would be better than anchors?

**Pull request merged**
